bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/handler/DLinkSmartPlugHandler.java

```diff
@@ -0,0 +1,152 @@
+                        logger.error("Unexpected command type for channel '{}'.", DLinkSmartHomeBindingConstants.STATE);
+                    }
+                }
+            }
+
+            @Override
+            protected void updateStatus(ThingStatus status, ThingStatusDetail detail, String message) {
+                DLinkSmartPlugHandler.this.updateStatus(status, detail, message);
+            }
+        };
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        hnap.stop();
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        hnap.handleCommand(channelUID, command);
+    }
+
+    @Override
+    public void initialize() {
+        final DLinkThingConfig config = getConfigAs(DLinkThingConfig.class);
+        hnap.setPin(config.pin);
+        hnap.setIp(config.ipAddress);
+        hnap.start(scheduler);
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        hnap.reset();
+    }
+
+    private State getCurrentConsuption() {
+        return hnap.queryUnitState("GetCurrentPowerConsumption", "CurrentConsumption", "2", SmartHomeUnits.WATT);
+    }
+
+    private State getTotalConsumption() {
+        return hnap.queryUnitState("GetPMWarningThreshold", "TotalConsumption", "2", SmartHomeUnits.KILOWATT_HOUR);
+    }
+
+    private State getTemperature() {
+        return hnap.queryUnitState("GetCurrentTemperature", "CurrentTemperature", "3", SIUnits.CELSIUS);
+    }
+
+    private State getState() {
+        return hnap.queryOnOffState("GetSocketSettings", "OPStatus", "1");
+    }
+
+    private void setState(OnOffType state) {
+        hnap.setState("SetSocketSettings", "SetSocketSettingsResult", "1", "Socket 1", "Socket 1", state);
+    }
+
+    private ChannelUID getChannelUID(final String name) {
+        Channel channel = thing.getChannel(name);
+        if (channel == null) {
+            logger.error("Did not find channel '{}'.", name);
+            return null;
+        } else {
+            return channel.getUID();
+        }
+    }
+
+    private State getNewChannelState(final String channelID) {
+        if (thing.getStatus().equals(ThingStatus.ONLINE)) {
+            switch (channelID) {
+                case DLinkSmartHomeBindingConstants.CURRENT_CONSUMPTION:
+                    return getCurrentConsuption();
+                case DLinkSmartHomeBindingConstants.TOTAL_CONSUMPTION:
+                    return getTotalConsumption();
+                case DLinkSmartHomeBindingConstants.TEMPERATURE:
+                    return getTemperature();
+                case DLinkSmartHomeBindingConstants.STATE:
+                    return getState();
+                default:
+                    logger.error("Unexpected channel: '{}'.", channelID);
+                    throw new IllegalArgumentException("Unexpected channel.");
```

Comment:
If you thrown the exception no need to also log. Pass the channel in the exception message:
```suggestion
                    throw new IllegalArgumentException("Unexpected channel: " + channelID);
```